User_Story#1/SortingApp/SortinfApp.UnitTests/CollectionSortHandlerTest.cs

```diff
@@ -0,0 +1,388 @@
+﻿using System;
+using System.Linq;
+using BusinessLayer;
+using BusinessLayer.Reader.Fakes;
+using BusinessLayer.Utilities.Parser.Interfaces.Fakes;
+using BusinessLayer.Utilities.Validator.Interfaces.Fakes;
+using BusinessLayer.Writer.Fakes;
+using Helper;
+using Microsoft.QualityTools.Testing.Fakes;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SortinfApp.UnitTests
+{
+    /// <summary>
+    /// Represents tests for CollectionSortHandler class.
+    /// </summary>
+    /// <owner>Anton Petrenko</owner>
+    [TestClass]
+    public class CollectionSortHandlerTest
+    {
+        /// <summary>
+        /// Collections the sort handler collection sort handler pass null value in constructor throw argument null exception.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void CollectionSortHandler_CollectionSortHandler_PassNullParserInConstructor_ThrowArgumentNullException()
+        {
+            //
+            // Arrange.
+            //
+            var reader = new StubDataReader(new StubIValidator());
+            var writer = new StubDataWriter<double>();
+
+            //
+            // Assert.
+            //
+            Assert.ThrowsException<ArgumentNullException>(() => new CollectionSortHandler<double>(reader, writer, null));
+        }
+
+        /// <summary>
+        /// Collections the sort handler collection sort handler pass null value in constructor throw argument null exception.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        [TestMethod]
+        public void CollectionSortHandler_CollectionSortHandler_PassNullReaderInConstructor_ThrowArgumentNullException()
+        {
+            //
+            // Arrange.
+            //
+            var parser = new StubIDataParser<double>();
+            var writer = new StubDataWriter<double>();
+
+            //
+            // Assert.
+            //
+            Assert.ThrowsException<ArgumentNullException>(() => new CollectionSortHandler<double>(null, writer, parser));
```

Comment:
The same about null.
Don't hesitate to add variable with name and initialize it with null.